README.md

```diff
@@ -679,6 +679,61 @@ It also means that these rules will match against following two events :
 
 This behaviour may change in future version (to avoid any confusions) and should not be relied upon.
 
+#### Caveat: JSON parsing and duplicate keys
+
+Events or Rules containing duplicate JSON keys are invalid. When duplicate keys are passed Ruler 
+will only consider the final value in the document order. 
+
+This means following two rules will compile to the same internal representation.
+
+```javascript
+## has duplicate keys
+{
+  "source": ["aws.s3"],
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail":  {
+      "eventSource": ["s3.amazonaws.com"],
+      "eventSource": ["sns.amazonaws.com"]
+  }
+}
+
+## has unique keys
+{
+  "source": ["aws.sns"],
+  "detail-type": ["AWS API Call via CloudTrail"],
+  "detail": { "eventSource": ["sns.amazonaws.com"] }
+}
+```
+
+Similarly, following two *events* will be treated the same
```

Comment:
Haha, when I was thinking about this I realized I didn't know what would actually happen.  I'm going to assume you tested this and verified that the first event will *not* match 
```json
{"source": [ "aws.s3"]}
```

Because I'm worried that it might.

Reply:
I did test this, just forgot to push the latest commit with the update. Also as the assertions you mentioned below.